File: 프로그래머스/2/42587. 프로세스/프로세스.java
import java.util.*;

class Solution {
    public int solution(int[] priorities, int location) {
        Queue<int[]> q = new LinkedList<>();
        Queue<Integer> pq = new PriorityQueue<>((a, b) -> b-a);
        
        for (int i = 0; i < priorities.length; i++) {
            q.add(new int[]{priorities[i], i});
            pq.add(priorities[i]);
        }
        
        int answer = 1;
        while(!q.isEmpty()) {

            if (q.peek()[0] == pq.peek()) {
                pq.poll();
                int[] qp = q.poll();
                if (qp[1] == location) {
                    return answer;
                }
                answer++;
            } else {
                q.add(q.poll());    
            }
            
        }
        return -1;
    }
}

File: 프로그래머스/2/42587. 프로세스/프로세스.py
from collections import deque
def solution(priorities, location):
    
    
    s_p = deque(sorted(priorities, reverse=True))
    
    
    priorities = list(enumerate(priorities))

    answer = 0
    while True:
        for i in range(len(priorities)):
            if priorities[i][1] == s_p[0]:
                s_p.popleft()
                answer += 1
                if priorities[i][0] == location:
                    return answer
